### PR TITLE
Kubeflow support for LLM

### PR DESCRIPTION
**Pull request by gavrishp**
Changes
- Introduce Kubeflow support for kubernetes deployment of LLMs

### CODE REVIEW

File: llm/run.sh [hunk 1]
```diff
@@ -93,23 +92,92 @@ function inference_exec_vm(){
...
+        echo "NFS info not provided"
+        helpFunction
+    fi
+
+    if [ -z $deploy_name ] ; then
+        echo "deployment metadata name not provided"
+        helpFunction
+    fi
+
+    mkdir $mount_path/$model_name/config
```
Comment (johnugeorge) [inline llm/run.sh:124]: does it error out if directory already present?

File: llm/run.sh [hunk 2]
```diff
@@ -93,23 +92,92 @@ function inference_exec_vm(){
...
+    cp $wdir/config.properties $mount_path/$model_name/config/
+
+    export INGRESS_HOST=$(kubectl get po -l istio=ingressgateway -n istio-system -o jsonpath='{.items[0].status.hostIP}')
```
Comment (johnugeorge) [inline llm/run.sh:127]: Are these vars used?

File: llm/run.sh [hunk 3]
```diff
@@ -93,23 +92,92 @@ function inference_exec_vm(){
...
+        k ) stop_server=0 ;;
+        a ) mar_file_path="$OPTARG" ;;
+        m ) mount_path="$OPTARG" ;;
+        f ) nfs="$OPTARG" ;;
+        e ) deploy_name="$OPTARG" ;;
+        ? ) helpFunction ;; # Print helpFunction in case parameter is non-existent
+   esac
+done
+
+shift $((OPTIND-3))
```
Comment (johnugeorge) [inline llm/run.sh:155]: what is this used for?

File: llm/run.sh [hunk 4]
```diff
@@ -2,43 +2,42 @@
...
 
 function create_execution_cmd()
 {
     echo $model_name
     gen_folder="gen"
-    if [ $compute_choice = "vm" ] ;
-    then
-        cmd="python3 $wdir/torchserve_run.py"
-    else
-        cmd="python3 code/torchserve/torchserve_run.py"
```
Comment (johnugeorge) [inline llm/run.sh:39]: why was this added earlier

Comment (gavrishp) [reply llm/run.sh:39]: It was picked from old nvd

File: llm/run.sh (hunk 1 above)
Comment (gavrishp) [reply llm/run.sh:124]: nope, just says Already exists and continues

File: llm/run.sh [hunk 5]
```diff
@@ -93,23 +92,92 @@ function inference_exec_vm(){
...
+    fi
+
+    mkdir $mount_path/$model_name/config
+    cp $wdir/config.properties $mount_path/$model_name/config/
+
+    export INGRESS_HOST=$(kubectl get po -l istio=ingressgateway -n istio-system -o jsonpath='{.items[0].status.hostIP}')
+    export INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].nodePort}')
+
+    echo "Running the Inference script";
+    python3 $wdir/kserve_run.py --gpu $gpus --cpu 8 --mem 32Gi --model_name $model_name --mount_path $mount_path --nfs $nfs --deploy_name $deploy_name --data $data
```
[the remark this answers is not in the document]
Comment (gavrishp) [reply llm/run.sh:131]: As param into run.sh?

File: llm/run.sh (hunk 2 above)
Comment (gavrishp) [reply llm/run.sh:127]: dont think so, this is the suggested approach in kserve docs
